demo/3000-home/next.config.js

```diff
@@ -1,20 +1,29 @@
 const NextFederationPlugin = require('@module-federation/nextjs-mf');
-
+const {promiseTemplate} = require('@module-federation/nextjs-mf/lib/build-utils');
 module.exports = {
   webpack(config, options) {
     const { isServer } = options;
     if (!options.isServer) {
+      const remotes = {
+        shop: `shop@http://localhost:3001/_next/static/${
+          isServer ? 'ssr' : 'chunks'
+        }/remoteEntry.js`,
+        checkout: `checkout@http://localhost:3002/_next/static/${
+          isServer ? 'ssr' : 'chunks'
+        }/remoteEntry.js`,
+      }
       config.plugins.push(
         new NextFederationPlugin({
           name: 'home_app',
           filename: 'static/chunks/remoteEntry.js',
           remotes: {
-            shop: `shop@http://localhost:3001/_next/static/${
-              isServer ? 'ssr' : 'chunks'
-            }/remoteEntry.js`,
-            checkout: `checkout@http://localhost:3002/_next/static/${
-              isServer ? 'ssr' : 'chunks'
-            }/remoteEntry.js`,
+            shop: promiseTemplate(remotes.shop, function() { new Promise((resolve, reject) => {
```

Comment:
Need improve this so we dont have to wrap in function call. 

Perhaps use generator / templator factory for creating promises internally